src/components/AppContext/AppContext.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { Todo } from '../../types/Todo';
+import { FilterMode } from '../../types/Filter';
+
+interface IContextValue {
+  userId: number,
+  allTodos: Todo[],
+  setAllTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  activeTodos: Todo[],
+  completedTodosIds: number[],
+  tempTodo: Todo | null,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>,
+  shouldShowError: boolean,
+  setShouldShowError: React.Dispatch<React.SetStateAction<boolean>>,
+  errorMessage: string,
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>,
+  showError: (errorText: string) => void,
+  loadingTodosIds: number[],
+  setLoadingTodosIds: React.Dispatch<React.SetStateAction<number[]>>,
+  currentFilterMode: FilterMode,
+  setCurrentFilterMode: React.Dispatch<React.SetStateAction<FilterMode>>,
+}
+
+export const AppContext = React.createContext<IContextValue>({
+  userId: 0,
+  allTodos: [],
+  setAllTodos: () => {},
+  activeTodos: [],
+  completedTodosIds: [],
+  tempTodo: null,
+  setTempTodo: () => {},
+  shouldShowError: false,
+  setShouldShowError: () => {},
+  errorMessage: '',
+  setErrorMessage: () => {},
+  showError: () => {},
+  loadingTodosIds: [0],
+  setLoadingTodosIds: () => {},
+  currentFilterMode: FilterMode.All,
+  setCurrentFilterMode: () => {},
+});
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const AppProvider: React.FC<Props> = ({ children }) => {
+  const userId = useMemo(() => 6826, []);
```

Comment:
Why do you need useMemo here? This is a constant value and better to move it outside